quantizers/uniform.py

```diff
@@ -59,9 +63,27 @@ def quantize(self, X):
             return np.zeros(X.shape) + center, total_bytes
 
         # Actually quantize.
-        compressed_X = _quantize(X_recentered, self.num_bits, sf)
+        compressed_X = self._quantize(X_recentered, self.num_bits, sf)
 
         # Recenter and return.
         compressed_X += center
 
         return compressed_X, total_bytes
+
+
+class MidtreadQuantizer(UniformQuantizer):
+    def name(self):
+        return "uniform_mt" + str(self.num_bits) + "b"
+
+    def quantize(self, X):
+        if (self.num_bits <= 1):  #mid-tread requires at least 2 bits
+            return np.zeros(X.shape), UniformQuantizer.get_total_bytes(self, X)
+
+        L = 2**self.num_bits - 1
+        eps = 1e-7
+        a = max(np.max(X), -1 * np.min(X)) + eps
+        delta = 2 * a / (L - 1)
```

Comment:
Delete the L-1 and just do it when you declare L above. Now if a is the negative number this is not 100% correct though. If a is the negative number there should be no -1 on the L. You should comment your code...